### PR TITLE
Dao transport branch

### LINKED ISSUE #10: Plan Python Dao Communication Protocol
Instances of the Dao application need to be able to talk to each other.  We will achieve this using socket programming with one instance of the program acting as a host.  This allows the programs to send data to each other.  This communication needs to be structured so that both instances of the program can recognize and utilize this data.

Objective:

- Create a list of message codes that can be exchanged 
- Define what each message code does
- Define any payload associated with each message
    - Some payloads may need to be placeholders for now, we can define objects for them later
- Create a basic message structure that all messages will follow

Points:

- This should all be wrapped up in a python file
- Keep all structures clean and expandable, we may want to add new messages later when we realize we missed something
- Do not implement functions to actually send any information for this task, just the structures and object definitions

### PR DESCRIPTION
**Pull request by donny138**
This PR contains two things:

- DaoGameFlow.txt
    - This file outlines the rough behavior / interaction of the client and server dao programs
    - This is mostly here for brainstorming, may need to be changed or modifed
- DaoMsgDef.py
    - This file implements a rough outline for a communication protocol.  See more below
    - This file implements the tasks for [DaoGame #10 ](https://github.com/milkshake-moo/Dao_Game/issues/10)
    - There is some temporary code within this file that will need revision / removal before full implementation is ready

Note:  This PR does not add anything involving exchanging data using sockets.  Just the definitions and objects related to different types of messages these programs may need to exchange in the future.

The following stuff now exists in the Dao message stuffs! 

- Message codes for different messages
    - These have rough payloads associated with them

- A Standard Dao Message object
    - stores everything associated with messages to exchange
    - can be created from just a series of bytes (what sockets actually exchange info using)

- Message constructors
    - creates a Standard Dao Message for a particular message code
    - takes any payload data for it's message code as input and adds it as bytes to the message object

- Message extractors
    - extracts payload data from a Standard Dao Message into normal variables types

- TEMPORARY MAIN FUNC:
    - This will be removed at some point
    - Contains a bunch of test / example code using everything above

### CODE REVIEW

File: Backbone/DaoMsgDef.py [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+    win_player = player_bytes.decode()
+    win_method = win_bytes.decode()
+    return end_status, win_player, win_method
+
+
+
+
+
+
+# ========================= TEMP TEST CODE PLZ IGNORE ==============================
```
Comment (milkshake-moo) [inline Backbone/DaoMsgDef.py:439]: noting existence of test code

**Review by milkshake-moo — APPROVED** [review]
looks great! I wish I understood more about the communication between the servers but this looks like it is definitely on the right track!

File: Backbone/DaoMsgDef.py (hunk 1 above)
Comment (donny138) [reply Backbone/DaoMsgDef.py:439]: The only way the donny knows how to program is trial and error lol

**Comment by donny138** [comment]
@milkshake-moo There isn't any actual communication yet!  Think of the message stuff as the words the future programs will use to communicate!   

The next step ( #11 ) is to write some scripts that can use these words to talk to each other.

**Comment by milkshake-moo** [comment]
@donny138 That's what I figured. Having all of the communication "types" and messages defined here will make the actual implementation easy, I think/hope!

**Pull request merged**